src/bootloaders/grub2.c

```diff
@@ -80,130 +100,284 @@ static inline char *grub2_get_boot_relative(void)
 
 bool grub2_init(__cbm_unused__ const BootManager *manager)
 {
+        kernel_queue = nc_array_new();
+        if (!kernel_queue) {
+                DECLARE_OOM();
+                abort();
+        }
         return true;
 }
 
 void grub2_destroy(__cbm_unused__ const BootManager *manager)
 {
+        if (kernel_queue) {
+                /* kernels pointers inside are not owned by the array */
+                nc_array_free(&kernel_queue, NULL);
+        }
 }
 
-bool grub2_install_kernel(const BootManager *manager, const Kernel *kernel)
+/**
+ * Push a pointer to the kernel into our queue for processing during set_default
+ */
+bool grub2_install_kernel(__cbm_unused__ const BootManager *manager, const Kernel *kernel)
```

Comment:
Yes - that's what the function pointer definition is for the bootloader struct `>install_kernel`